src/cli/commands/public-cmds/show-cmd.js

```diff
@@ -113,6 +128,12 @@ export default class Show extends Command {
       const componentFromFileSystem = makeComponentReadable(component);
       const componentFromModel = makeComponentReadable(componentModel);
       const jsonObject = componentFromModel ? { componentFromFileSystem, componentFromModel } : componentFromFileSystem;
+
+      if (specifickeys && json) {
```

Comment:
I would move this logic into consumerComponent.toObject

Reply:
https://github.com/teambit/bit/blob/master/src/consumer/component/consumer-component.js#L775